sdk/internal/testframework/request_matcher.go

```diff
@@ -0,0 +1,111 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package testframework
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+
+	"github.com/dnaeon/go-vcr/cassette"
+)
+
+type RequestMatcher struct {
```

Comment:
How is this used, or is it vestigial and can be removed?

Reply:
It's used to provide custom logic for how requests should be matched.

Reply:
OK, yeah it has been a while since I have looked at this :) 
You're correct - I think in its current state, it only serves as the default implementation for matching logic. We exported the Sanitizer on `Recording` but never did the same for the RequestMatcher. I can do that now to allow for the extensibility, or we can let it evolve as needed. I'm fine to make the changes if we want to export it now.

Reply:
My default statement on such things is to remove them, and add them if/when there's an actual need for it.  If @mohsha-msft can comment if `azblob` needs this functionality then let's keep it and ensure it works.

Reply:
It's definitely needed, the question is whether or not we expose the extensibility.

Reply:
I think I'll go ahead and implement the changes, as I am pretty sure we'll need them.